src/ssl.c

```diff
@@ -29472,6 +29488,20 @@ int wolfSSL_SSL_in_init(WOLFSSL *ssl)
     return !wolfSSL_is_init_finished(ssl);
 }
 
+int wolfSSL_SSL_in_before(const WOLFSSL *ssl)
+{
+    WOLFSSL_ENTER("wolfSSL_SSL_in_before");
+
+    if (ssl == NULL)
+        return WOLFSSL_FAILURE;
+
+    if (ssl->options.side == WOLFSSL_CLIENT_END) {
+        return ssl->options.connectState == CONNECT_BEGIN;
+    }
+    return ssl->options.acceptState == ACCEPT_BEGIN ||
+        ssl->options.acceptState == TLS13_ACCEPT_BEGIN;
+}
```

Comment:
Using `ssl->options.connectState` and `ssl->options.acceptState` has been a source of bugs in the past because they mean different things in TLS 1.3 and <= 1.2. All the `*BEGIN` values evaluate to `0` but I would prefer if this API would just use `return ssl->options.handShakeState == NULL_STATE;` instead. This would also not require the need to branch on the connection side.

Reply:
Updated to fix this.  I did notice the existing API wolfSSL_SSL_in_connect_init is using connect/acceptState as well, does this function need to be fixed?

Reply:
Yes `wolfSSL_SSL_in_connect_init` should also be modified to use `ssl->options.handShakeState`. Can you modify it to be a copy of `wolfSSL_SSL_in_init`?

Reply:
Updated per your request, but are you sure this is correct?
It looks like wolfSSL_in_connect_init is supposed to only return 1 if we are acting as a client and wolfSSL_in_init would return 1.

Reply:
Let's revert this change for now and we can address these functions in a separate PR. It should check if we are in progress of handshaking.

Reply:
Done